Fix russian translation and add number translation use regexp (lua pattern)

il-s opened the pull request:
Hello from Russia!

Your translation into Russian is terrible. We fixed most of the translation errors and supplemented what you have in get_text, but it is not in the translations table (in the database / additional_translation.sql file).

Your mechanism for translating numbers is very simple and, as I understand it, is based on adding and removing prefixes. However, in everyday work, very often, you need to do something more complex. The use of regular expressions (even greatly simplified as in LUA) gives much more opportunities. For example, you want to check that the number contains 11 digits and starts at 79 (RUSSIA mobile), then writing in the CUT field: "LP: ^ (79 .........) $", in the ADD field: "888% 1 ", we can add a special prefix for routing. Checking the length will save us from problems if an internal number starts at 79 or for example a local number (city numbers of 6 digits). I understand that our method is far from ideal, but perhaps it will be useful for someone and will help you make a more general solution.

mihalevm committed:
Added LUA pattern number translations

mihalevm committed:
Russian localization

mihalevm committed:
DB: add some module_name items into translations table

mihalevm committed:
Translations

mihalevm committed:
Fixed bug: Wrong localization numbers with float point in different language

smrdoshi commented on freeswitch/scripts/astpp/lib/astpp.functions.lua:
```diff
@@ -281,6 +281,18 @@ function do_number_translation(number_translation,destination_number)
 	return destination_number
       end
+
+      if (string.sub(tmp_str[1],0,3) == 'LP:') then
```
Hi @mihalevm @il-s 

Thanks for your contribution. 

We would like to understand this condition code. Can you please explain with example? 
I appreciate your efforts.

il-s replied:
Example in request text:
"For example, you want to check that the number contains 11 digits and starts at 79 (RUSSIA mobile), then writing in the CUT field: "LP: ^(79.........)$", in the ADD field: "888%1", we can add a special prefix for routing. Checking the length will save us from problems if an internal number starts at 79 or for example a local number (city numbers of 6 digits)."

The prefix "LP:" (Lua Pattern) allows you to separate the normal behavior during translation and translation using regular expressions.

smrdoshi commented on web_interface/astpp/application/libraries/csvreader.php:
```diff
@@ -103,7 +103,8 @@ function parse_lines($p_CSVLines, $config_variable, $check_header_flag = false)
 					$value = preg_replace ( '#<script.*</script>#is', '', $value );
 					if (isset ( $field_key_value ) && ! empty ( $field_key_value ))
-						$content [$field_key_value] = strip_tags ( filter_var ( $value, FILTER_SANITIZE_STRING ) );
+//						$content [$field_key_value] = strip_tags ( filter_var ( $value, FILTER_SANITIZE_STRING ));
```
We need this line here as some time we are getting html text in value. So we want to clean them before processing.

mihalevm replied:
Some translations has html tags (for examle "Connection \<br\>Cost"). Installing laguage addon pass througth this module and filtering some strings like this. I think this bad practice, used html tags in translation, better use css-style for word wrap in tables header. But now for minimum code correction remove this filter from csvreader library.

smrdoshi commented on web_interface/astpp/application/views/header.php:
```diff
@@ -141,6 +141,7 @@ function process_alert(id,table) {
      <script src="<?php echo base_url(); ?>assets/js/jquery-1.12.4.js"></script>
      <script type="text/javascript" src="<?php echo base_url(); ?>assets/js/bootstrap.bundle.min.js"></script>
 	 <script type="text/javascript" src="<?php echo base_url(); ?>assets/js/bootstrap-select.js"></script>  
+     <script type="text/javascript" src="<?php echo base_url(); ?>assets/js/i18n/defaults-<?=$_COOKIE["lang"]?>.js"></script>
```
Please explain the usage of this js?

mihalevm replied:
Project include JS libraries, such as datetimepicker, bootstrap-select. Some libraries already has translation. For examle bootstrap-select library has own localization (path in project: /web_interface/astpp/assets/js/i18n/defaults-xx_XX.js). This visual component used in some places, example (Services>Product>Edit Product>Add Destination>"Call Type, Rate Group, Country"). By using cookie variable "lang" (ru_RU) and same part of file (defaults-ru_RU.js) we can loading current translation for some JS libraries.

![изображение](https://user-images.githubusercontent.com/46156939/67914548-1457be80-fbc3-11e9-8ea1-a6be0840f1a6.png)

mihalevm committed:
Fix problem use symbol '=' on Gateway\Dialplan Variable
Example: try add "nolocal:sip_invite_params=user=phone" for add to request line ";user=phone"

mihalevm committed:
Russian translation update

mihalevm committed:
Additional Russian translation

mihalevm committed:
Fix: Wrong call destination name convertation in cdr loger. Fix sub-admin permission for addon fsmonitor.

mihalevm committed:
Add Russain localization for addon FS_Monitor

mihalevm committed:
Add Russain localization Grid generator. Fix bug with action buttons.

mihalevm committed:
Add Russain localization. Fix bug with build_form function. Change time duration format on user dashboard.

mihalevm committed:
Fix bug with build_grid function.

mihalevm committed:
Fix: CreditLimit not set. Wrong currency data.

mihalevm committed:
Localize invoice template

mihalevm committed:
Full rewrite func do_number_translation: use ONLY lua pattern, use match with MAX length, fixed stupid bugs

mihalevm committed:
Cleanup do_number_translation, delete compat code

mihalevm committed:
Fixed translation CallerID in outbound calls

mihalevm committed:
Add ASTPP firewall control

mihalevm committed:
Fix Russian localization

mihalevm committed:
Fix Russian localization

mihalevm committed:
Disabled user tracking. Ignore logging web session for some ip.

mihalevm committed:
Added ip-map typization: acl-rule, fw-rule

mihalevm committed:
Add Customer note. Add in summry report trunk filter.

mihalevm committed:
Change Documentation link in main menu.

il-s committed:
Add random translation in Localizations

il-s committed:
Add bank payments

il-s committed:
Add bank payments controller

il-s committed:
Add in trunk and products select calltype. Show in user products remain minutes.

il-s committed:
Fixes for previous commit.

il-s committed:
Add colums Free minutes and Remain minutes in client UI

il-s committed:
Fixed package calc with numbers is not E.164 (problem package_calc before number translations)
Fixed save to CDR numbers, before number translations

il-s committed:
Add Yandex.Money payment gateway

il-s committed:
Add Yandex.Money payment gateway

il-s committed:
Bug: translation in used method

il-s committed:
Bug: Select user account in customer report

il-s committed:
Bug: Translationin some places

il-s committed:
Add filter for client cdrs into admin panel

il-s committed:
Fix bug CDRs table header

il-s committed:
Fix bug CDRs table header 2

il-s committed:
Fix bug CDRs table header 3

il-s committed:
Fix bug CDRs table header 4

sakhnyuk committed:
Add logo

sakhnyuk committed:
Update Footer and Header

sakhnyuk committed:
Update logo

il-s committed:
Reworked architecture for issuing conversations

il-s committed:
Record files. Changed directory path

il-s committed:
Rework and fix some bugs for Record files

il-s committed:
Bug in Assign Packet. Wrong price number format

il-s committed:
Add display policy for client call records

il-s committed:
Bug: Display button play when record absent

il-s committed:
RateDeck rework

il-s committed:
Add filter item in CDRs report. Fix translation bug.

il-s committed:
YM Fix bug with instance_id

il-s committed:
Add report by direction

il-s committed:
Fix default value in ip_map module

mihalevm committed:
Design view reworked

mihalevm committed:
Packet Analytics: Add admin part
SupportTicket: Fix menu in user cab

mihalevm committed:
Packet Analytics: Fix bug

mihalevm committed:
Packet Analytics: Fix bug 2

mihalevm committed:
Packet Analytics: Fix bug 2

mihalevm committed:
Packet Analytics: Optimize SQL
Main menu: Fix subAdmin menu

mihalevm committed:
Optimization: User CDRs report

mihalevm committed:
Fix bug: Save products

mihalevm committed:
Add product: vPBX

mihalevm committed:
Add: Search filter by Trunks in User CDRs Report

mihalevm committed:
Add into CDRs and Reseller_CDRs answersec field

mihalevm committed:
Add Column Answer duration in CDRs report

il-s committed:
Add macro %%diversion in gateway dialplan variables. For use put string like %%diversion=<sip:%%diversion@>;user=phone|3 (3 mean localization index)

il-s committed:
Merge branch 'v4.0.1' of https://github.com/mihalevm/ASTPP into v4.0.1

il-s committed:
Fixed macro %%diversion: use wrong callerid and delete debug comments
